.github/workflows/check_changelog.yml

```diff
@@ -0,0 +1,33 @@
+name: Check Changelog
+# This check makes sure that the changelog is properly updated
+# when a PR introduces a change in a test file.
+# To bypass this check, label the PR with  no-changelog.
+on:
+  pull_request:
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'no-changelog') == 0 }}
+    steps:
+      - name: Get PR number
+        run: |
+          echo "PR_NUMBER=${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: '0'
+      - name: Check the changelog only if tests have been modified
+        run: |
+          set -xe
+          tests=$(git diff --name-only origin/master)
+          if [[ "$tests" =~ tests ]]
+          then
+            result=$(cat ./doc/whats_new/v*.rst)
+            if [[ "$result" =~ :pr:\`$PR_NUMBER\` ]]
+            then
+              echo "Changelog has been updated."
+            else
+              echo "Changelog entry is missing. If no changelog entry is required for this PR, label the PR with no-changelog to bypass this check."
```

Comment:
Maybe we should be extra-explicit on case where we need a changelog entry:

- bug fixes
- new features (in particular with changes to the public API, e.g. with a new public class or function or a new parameter)

Changes that do not need a changelog entry:

- documentation fixes or improvements
- new or updated examples or benchmark
- fix or improvements in the test suite
- code style changes
- updated continuous integration configuration
- updated packaging configuration or tools (unless the previously released packages on pypi.org or conda-forge where invalid for some reason, in which case it's useful to mention the list of problematic packages in the changelog)

Reply:
>     * fix or improvements in the test suite

wrt the current implementation this case will be checked because test files are likely to be modified. Should I get rid of this check?
https://github.com/scikit-learn/scikit-learn/blob/62a506861e9c16348764e5187da6749ac3402312/.github/workflows/check_changelog.yml#L22-L23

> Note that it is ok to reference several PRs in the same changelog entry, for instance when a big task has been decomposed in subtasks or if a partial fix has been further improved in a follow-up PR.

The current implementation check if the submitted PR is cited in the changelog: this is always true when a changelog is needed, right?

Reply:
If the PR is only a about fixing a bad test or adding a new one, we shall not document it in a changelog entry because it's probably useless for our end-users. I mean the test fix is useful to us and our users but since the tests are not part of the public API, it's not that interesting to the readers of the changelog.

> The current implementation check if the submitted PR is cited in the changelog: this is always true when a changelog is needed, right?

Yes, this is correct.